### PR TITLE
new resource: azurerm_kusto_iothub_data_connection

### PR DESCRIPTION
**Pull request by njuCZ**
refactor kusto RP and add new resource
image

### CODE REVIEW

File: azurerm/internal/services/kusto/tests/kusto_iothub_data_connection_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+)
+
+func TestAccAzureRMKustoIotHubDataConnection_basic(t *testing.T) {
```
Comment (katbyte) [inline azurerm/internal/services/kusto/tests/kusto_iothub_data_connection_resource_test.go:11]: Could we add a complete test which sets _all_ optional properties and then an update test that goes basic -> complete -> basic?

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @njuCZ - overall this looks good but could we get a complete and update test? thanks!

**Comment by njuCZ** [comment]
@katbyte complete test is hard for this resource, because the optional fields "table_name", "mapping_rule_name" are kusto RP data plane resource. Fow now, we could not create them through terraform. and I found existing resource `azurerm_kusto_eventhub_data_connection` keeps these fields and no complete acctest.

Do you think we would better delete these fields or just keep them and no test?

**Comment by katbyte** [comment]
@njuCZ - can we just delete the properties we cannot test and leave the rest for now? thanks

**Comment by njuCZ** [comment]
@katbyte Thanks for your advice, I have deleted these fields. Please have a look again when free.

**Comment by njuCZ** [comment]
just rebased the latest master branch and resolve conflicts

**Comment by katbyte** [comment]
@njuCZ - this is failing to build with:
```
[15:38:08]	[Step 4/5] ./kusto_cluster_customer_managed_key_resource.go:11:2: imported and not used: "github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
[15:38:08]	[Step 4/5] ./kusto_eventhub_data_connection_resource.go:16:2: validate redeclared as imported package name
[15:38:08]	[Step 4/5] 	previous declaration at ./kusto_eventhub_data_connection_resource.go:14:2
[15:38:08]	[Step 4/5] ./kusto_eventhub_data_connection_resource.go:90:19: undefined: "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/kusto/validate".ValidateEventHubConsumerName
[15:39:30]	[Step 4/5] Process exited with code 2
[15:39:30]	[Step 4/5] Process exited with code 2 (Step: Compile Test Binary (Command Line))
[15:39:30]	[Step 4/5] Step Compile Test Binary (Command Line) failed
[15:39:30]	[Step 4/5] Error message is logged
```

**Comment by njuCZ** [comment]
@katbyte I have rebased this PR, now it could build successfully and acctest could pass.
image

**Review by katbyte — APPROVED** [review]
Thanks @njuCZ - LGTM 👍

**Pull request merged**